starfish/test/test_overlap_utils.py

```diff
@@ -0,0 +1,157 @@
+    it3 = create_intenisty_table_with_coords(Area(min_x=0, max_x=1,
+                                                  min_y=1, max_y=2))
+    overlaps = find_overlaps_of_xarrays([it0, it1, it2, it3])
+    # should have 4 total overlaps
+    assert len(overlaps) == 4
+    # overlap 1 between it0 and it1:
+    overlap_1 = Area(min_x=.5, max_x=1, min_y=.5, max_y=1)
+    assert overlap_1 == overlaps[(0, 1)]
+    # overlap 2 between it1 and it2
+    overlap_2 = Area(min_x=1.5, max_x=2, min_y=.5, max_y=1)
+    assert overlap_2 == overlaps[(1, 2)]
+    # overlap 3 between it1 and it3
+    overlap_3 = Area(min_x=.5, max_x=1, min_y=1, max_y=1.5)
+    assert overlap_3 == overlaps[(1, 3)]
+    # overlap 4 between it0 and it3
+    overlap_4 = Area(min_x=0, max_x=1, min_y=1, max_y=1)
+    assert overlap_4 == overlaps[(0, 3)]
+
+
+def test_remove_area_of_xarray():
+    """
+    Tests removing a section of an IntensityTable defined by its physical area
+    """
+    it = create_intenisty_table_with_coords(Area(min_x=0, max_x=2,
+                                                 min_y=0, max_y=2), n_spots=10)
+
+    area = Area(min_x=1, max_x=2, min_y=1, max_y=3)
+    # grab some random coord values in this range
+    removed_x = it.where(it.xc > 1, drop=True)[Coordinates.X.value].data[0]
+    removed_y = it.where(it.yc > 1, drop=True)[Coordinates.X.value].data[3]
+
+    it = remove_area_of_xarray(it, area)
+    # assert coords from removed section are no longer in it
+    assert not np.any(np.isclose(it[Coordinates.X.value], removed_x))
+    assert not np.any(np.isclose(it[Coordinates.Y.value], removed_y))
+
+
+def test_sel_area_of_xarray():
+    """
+    Tests selecting a section of an IntensityTable defined by its physical area
+    """
+    it = create_intenisty_table_with_coords(Area(min_x=0, max_x=2, min_y=0, max_y=2), n_spots=10)
+
+    area = Area(min_x=1, max_x=2, min_y=1, max_y=3)
+    it = sel_area_of_xarray(it, area)
+
+    # Assert new min/max values
+    assert min(it[Coordinates.X.value]).data >= 1
+    assert max(it[Coordinates.X.value]).data <= 2
+    assert min(it[Coordinates.Y.value]).data >= 1
+    assert max(it[Coordinates.X.value]).data <= 2
+
+
+def test_take_max():
+    """
+    Create two overlapping IntensityTables with differing number of spots and verify that
+    by concatenating them with the TAKE_MAX strategy we only include spots in the overlapping
+    section from the IntensityTable that had the most.
+    """
+    it1 = create_intenisty_table_with_coords(Area(min_x=0, max_x=2,
+                                                  min_y=0, max_y=2), n_spots=10)
+    it2 = create_intenisty_table_with_coords(Area(min_x=1, max_x=2,
+                                                  min_y=1, max_y=3), n_spots=20)
+
+    concatenated = IntensityTable.concatanate_intensity_tables(
+        [it1, it2], overlap_strategy=OverlapStrategy.TAKE_MAX)
+
+    # The overlap section hits half of the spots from each intensity table, 5 from it1
```

Comment:
wait what?  if it hits 5 of the spots from it1, then shouldn't we get a total of 25 spots?

Reply:
both the sel and remove_area_of_xarray methods are inclusive...so we get one spot in the comparison count and the concatenation...maybe this is wrong?

Reply:
I would dump the table to make sure it is consistent with your understanding, though I suspect you are correct. :)